### PR TITLE
Set dynamic datasource interval to default when collect rules are empty

### PR DESCRIPTION
**Pull request by subbarao-meduri**
Signed-off-by: Subbarao Meduri <smeduri@redhat.com>

### CODE REVIEW

File: operators/multiclusterobservability/controllers/placementrule/manifestwork.go [hunk 1]
```diff
@@ -598,6 +598,8 @@ func mergeCollectorRuleGroupList(defaultCollectRuleGroupList []operatorconfig.Co
 		}
 	}
 
+	config.CollectRulesEnabled = len(mergedCollectRuleGroups) > 0
```
Comment (marcolan018) [inline operators/multiclusterobservability/controllers/placementrule/manifestwork.go:601]: should be this? Any ootb group disabled, don't use the dynamic datasource?
```
config.CollectRulesEnabled = len(mergedCollectRuleGroups) == 2
```

Comment (subbarao-meduri) [reply operators/multiclusterobservability/controllers/placementrule/manifestwork.go]: Good point. Fixed in the latest commit.

File: operators/multiclusterobservability/controllers/multiclusterobservability/grafana.go [hunk 2]
```diff
@@ -74,7 +74,7 @@ func GenerateGrafanaDataSource(
 
 	DynamicTimeInterval := mco.Spec.ObservabilityAddonSpec.Interval
 
-	if DynamicTimeInterval > 30 {
+	if DynamicTimeInterval > 30  && !config.CollectRulesEnabled {
```
Comment (marcolan018) [inline operators/multiclusterobservability/controllers/multiclusterobservability/grafana.go:77]: should be
```
if DynamicTimeInterval > 30 && config.CollectRulesEnabled {
```

Comment (subbarao-meduri) [reply operators/multiclusterobservability/controllers/multiclusterobservability/grafana.go]: resolved.

**Pull request merged**